builder/src/non_permissioned.rs

```diff
@@ -0,0 +1,377 @@
+        .phrase::<&str>(&builder_params.mnemonic)
+        .index(builder_params.eth_account_index)?
+        .build()?;
+
+    tracing::info!("Builder account address {:?}", wallet.address());
+
+    let l1_client = L1Client::new(l1_params.url, Address::default());
+
+    let instance_state = NodeState::new(
+        l1_client,
+        wallet,
+        Arc::new(StatePeers::<Ver>::from_urls(network_params.state_peers)),
+    );
+    Ok(instance_state)
+}
+
+impl BuilderConfig {
+    pub async fn init(
+        pub_key: PubKey,
+        priv_key: PrivKey,
+        bootstrapped_view: ViewNumber,
+        channel_capacity: NonZeroUsize,
+        instance_state: NodeState,
+        hotshot_events_api_url: Url,
+        hotshot_builder_apis_url: Url,
+    ) -> anyhow::Result<Self> {
+        // tx channel
+        let (tx_sender, tx_receiver) = broadcast::<MessageType<SeqTypes>>(channel_capacity.get());
+
+        // da channel
+        let (da_sender, da_receiver) = broadcast::<MessageType<SeqTypes>>(channel_capacity.get());
+
+        // qc channel
+        let (qc_sender, qc_receiver) = broadcast::<MessageType<SeqTypes>>(channel_capacity.get());
+
+        // decide channel
+        let (decide_sender, decide_receiver) =
+            broadcast::<MessageType<SeqTypes>>(channel_capacity.get());
+
+        // builder api request channel
+        let (req_sender, req_receiver) = broadcast::<MessageType<SeqTypes>>(channel_capacity.get());
+
+        // builder api response channel
+        let (res_sender, res_receiver) = unbounded();
+
+        // create the global state
+        let global_state: GlobalState<SeqTypes> = GlobalState::<SeqTypes>::new(
+            (pub_key, priv_key),
+            req_sender,
+            res_receiver,
+            tx_sender.clone(),
+            instance_state.clone(),
+        );
+
+        let global_state = Arc::new(RwLock::new(global_state));
+
+        let global_state_clone = global_state.clone();
+
+        let builder_state = BuilderState::<SeqTypes>::new(
+            (
+                bootstrapped_view,
+                vid_commitment(&vec![], GENESIS_VID_NUM_STORAGE_NODES),
+                fake_commitment(),
+            ),
+            tx_receiver,
+            decide_receiver,
+            da_receiver,
+            qc_receiver,
+            req_receiver,
+            global_state_clone,
+            res_sender,
+            NonZeroUsize::new(1).unwrap(),
+            bootstrapped_view,
+        );
+
+        // create a client for it
+        // Start Client for the event streaming api
+        let client = Client::<EventStreamApiError, Version01>::new(hotshot_events_api_url.clone());
+
+        assert!(client.connect(Some(Duration::from_secs(60))).await);
+
+        tracing::info!("Builder client connected to the hotshot events api");
+
+        // client subscrive to hotshot events
+        let subscribed_events = client
+            .socket("hotshot_events/events")
```

Comment:
Minor modification: surf-disco default for streaming is json; for performance reasons, add this line:

```rust
            .socket("hotshot_events/events")
            .header(ACCEPT, "application/octet-stream")
            .subscribe::<BuilderEvent<SeqTypes>>()
```

Reply:
Done!